#! /bin/sh -e
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Now --help and --version exits with 0 instead of 1.

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0
@DPATCH@

diff -urN valgrind-2.2.0+2.4.0rc1.orig/coregrind/vg_main.c valgrind-2.2.0+2.4.0rc1/coregrind/vg_main.c
--- valgrind-2.2.0+2.4.0rc1.orig/coregrind/vg_main.c	2005-03-07 22:06:46.000000000 +0000
+++ valgrind-2.2.0+2.4.0rc1/coregrind/vg_main.c	2005-03-09 19:13:15.000000000 +0000
@@ -330,7 +330,7 @@
    VG_(message)(Vg_UserMsg,
       "");
    VG_(pp_sched_status)();
-   VG_(exit)(1);
+   VG_(exit)(0);
 }
 
 /* Debugging thing .. can be called from assembly with OYNK macro. */
